dace/transformation/helpers.py

```diff
@@ -17,6 +18,307 @@
 from dace.memlet import Memlet
 
 
+def nest_sdfg_subgraph(sdfg: SDFG,
+                       subgraph: SubgraphView,
+                       start: Optional[SDFGState] = None,
+                       name: Optional[str] = None) -> SDFGState:
+
+    # Nest states
+    states = subgraph.nodes()
+    if len(states) > 1:
+
+        if start is not None:
+            source_node = start
+        else:
+            src_nodes = subgraph.source_nodes()
+            if len(src_nodes) != 1:
+                raise NotImplementedError
+            source_node = src_nodes[0]
+
+        sink_nodes = subgraph.sink_nodes()
+        if len(sink_nodes) != 1:
+            raise NotImplementedError
+        sink_node = sink_nodes[0]
```

Comment:
in this case we can just make up a sink node

Reply:
This not so simple because you need to generate appropriate edges for the control flow following each node outside the SDFG subraph. This has to be studied further.